v2.0/inet.md

```diff
@@ -1,6 +1,86 @@
+<span class="version-tag">New in v2.0:</span> The `INET` [data type](data-types.html) stores an IPv4 or IPv6 address.
+
+<div id="toc"></div>
+
+## Syntax
+
+An `INET` value can be expressed using the following formats:
+
+Format | Description
+-------|-------------
+IPv4 | Standard [RFC791](https://tools.ietf.org/html/rfc791)-specified format of 4 octets expressed individually in decimal numbers and separated by periods. Optionally, the address can be followed by a subnet mask.<br><br> Examples: `'190.0.0.0'`, `'190.0.0.0/24'`
+IPv4-mapped IPv6 | IPv6 address mapped to a IPv4 address. <br><br>Example: `'::ffff:192.168.0.1/24'`
+IPv6 | Standard [RFC8200](https://tools.ietf.org/html/rfc8200)-specified format of 8 colon-separated groups of 4 hexadecimal digits. Optionally, the address can be followed by a subnet mask.<br><br> Example: `'2001:4f8:3:ba:2e0:81ff:fe22:d1f1'`, `'2001:4f8:3:ba:2e0:81ff:fe22:d1f1/120'`
+
+{{site.data.alerts.callout_info}}IPv4 addresses will sort before IPv6 addresses, including IPv4 addresses mapped to IPv6 addresses.{{site.data.alerts.end}}
+
+## Size
+
+An `INET` value is 32 bits for IPv4 or 128 bits for IPv6.
+
+## Example
+
+{% include copy-clipboard.html %}
+~~~ sql
+> CREATE TABLE computers (
+    ip INET PRIMARY KEY,
+    user_email STRING,
+    registration_date DATE
+  );
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SHOW COLUMNS FROM computers;
+~~~
+~~~
++-------------------+--------+-------+---------+-------------+
+|       Field       |  Type  | Null  | Default |   Indices   |
++-------------------+--------+-------+---------+-------------+
+| ip                | INET   | false | NULL    | {"primary"} |
+| user_email        | STRING | true  | NULL    | {}          |
+| registration_date | DATE   | true  | NULL    | {}          |
++-------------------+--------+-------+---------+-------------+
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> INSERT INTO computers
+  VALUES
+    ('192.168.0.1', 'info@cockroachlabs.com', '2018-01-31'),
+    ('192.168.0.2/10', 'lauren@cockroachlabs.com', '2018-01-31');
+    ('2001:4f8:3:ba:2e0:81ff:fe22:d1f1/120', 'test@cockroachlabs.com', '2018-01-31');
+~~~
+
+{% include copy-clipboard.html %}
+~~~ sql
+> SELECT * FROM computers;
+~~~
+~~~
++--------------------------------------+--------------------------+---------------------------+
+|                  ip                  |        user_email        |     registration_date     |
++--------------------------------------+--------------------------+---------------------------+
+| 192.168.0.1                          | info@cockroachlabs.com   | 2018-01-31 00:00:00+00:00 |
+| 192.168.0.2/10                       | lauren@cockroachlabs.com | 2018-01-31 00:00:00+00:00 |
+| 2001:4f8:3:ba:2e0:81ff:fe22:d1f1/120 | test@cockroachlabs.com   | 2018-01-31 00:00:00+00:00 |
++--------------------------------------+--------------------------+---------------------------+
+~~~
+
+## Supported Casting & Conversion
```

Comment:
@nvanbenschoten - similar to above, how do you determine what INET can be cast to?

Reply:
I think the easiest way is to look at the code: https://github.com/cockroachdb/cockroach/blob/94a2182a5e00fc0963c0f20c1dc278b12a9df7d2/pkg/sql/sem/tree/expr.go#L1254.

`inetCastTypes` is a list of all the types that INET can be cast to. It looks like only STRINGs can be cast to INET. I'll be auditing all of this as part of my QA process.

Reply:
Thanks!